stdlib/src/utils/string_slice.mojo

```diff
@@ -26,6 +26,129 @@ alias StaticString = StringSlice[ImmutableStaticLifetime]
 """An immutable static string slice."""
 
 
+fn is_valid_utf8[
+    width: Int = 8
```

Comment:
Q: did you do any benchmarking to determine that this should be 8 and not 4 or 16, or is this just an educated guess?

Reply:
Educated guess since: 
- the bigger the width the more bytes can be discarded at once when they are all ASCII
- 8x8 = 64 bits which fits into any SIMD engine. 
- I thought of adding 16 but I think I read somewhere that most small strings fall into 10 - 24 bytes long (might be wrong though and could make this bigger, will google some more later)

Reply:
ok, thanks!